test/unit/tuple_uint32_overflow.c

```diff
@@ -0,0 +1,77 @@
+#include "tuple.h"
+#include "unit.h"
+#include <stdio.h>
+#include <memory.h>
+#include <fiber.h>
+
+
+static char data[500];
+
+struct tuple *
+tuple_new_data()
+{
+	char* end = data;
+	end = mp_encode_array(end, 2);
+	end = mp_encode_uint(end, UINT32_MAX);
+	end = mp_encode_uint(end, UINT64_MAX);
+	struct tuple *tuple;
+	tuple = tuple_new(tuple_format_runtime, data, end);
+	tuple_ref(tuple);
+
+	return tuple;
+}
+
+
+static void
+tuple_next_u32_test()
+{
+	header();
+	plan(2);
+
+	struct tuple *tuple = tuple_new_data();
+	uint32_t field;
+	struct tuple_iterator it;
+	tuple_rewind(&it, tuple);
+	tuple_next_u32(&it, &field);
+	is(field, UINT32_MAX, "can read next uint32_t");
+	is(tuple_next_u32(&it, &field), -1, "can't read next uint64_t");
+	tuple_unref(tuple);
+
+	check_plan();
+	footer();
+}
+
+static void
+tuple_field_u32_test()
+{
+	header();
+	plan(2);
+
+	struct tuple *tuple = tuple_new_data();
+	uint32_t field;
+	tuple_field_u32(tuple, 0, &field);
+	is(field, UINT32_MAX, "can read uint32_t");
+	is(tuple_field_u32(tuple, 1, &field), -1, "can't read uint64_t");
+	tuple_unref(tuple);
+
+	check_plan();
+	footer();
+}
+
+int
+main()
```

Comment:
I've applied this test on master branch and test seems to pass. Please check it and in case this is true rewrite test so it fails without patch being applied.

Reply:
Strange. I also applied this test on master branch, but test didn't pass.